kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs

```diff
@@ -513,50 +512,40 @@ impl<'tcx> GotocCtx<'tcx> {
                 .collect()
         });
 
+        // Create the allocation from an array byte array.
+        let init_fn = |gcx: &mut GotocCtx, var: Symbol| {
+            let val = Expr::struct_expr_from_values(
+                alloc_typ_ref.clone(),
+                alloc_data
+                    .iter()
+                    .map(|d| match d {
+                        AllocData::Bytes(bytes) => Expr::array_expr(
+                            Type::unsigned_int(8).array_of(bytes.len()),
+                            bytes
+                                .iter()
+                                // We should consider adding a poison / undet where we have none
+                                // This mimics the behaviour before StableMIR though.
+                                .map(|b| Expr::int_constant(b.unwrap_or(0), Type::unsigned_int(8)))
```

Comment:
is it really the case that every individual byte is potentially undefined ?

Reply:
Because we don't care about the value and 0 is simpler. Reading uninitialized values is always UB, hence we are adding code to check for that.

Reply:
Do we have the guarantee that the check for uninitialised values isn't inhibited by this replacement of nondet by 0 ?

Reply:
The uninitialized value check should consider the type of the static. Is that the case today, @artemagvanian?

Reply:
I think it does if I understand you correctly: here is the code snippet for blessing statics:
```rust
fn visit_operand(&mut self, operand: &Operand, location: Location) {
    if let Operand::Constant(constant) = operand {
        if let ConstantKind::Allocated(allocation) = constant.const_.kind() {
            for (_, prov) in &allocation.provenance.ptrs {
                if let GlobalAlloc::Static(_) = GlobalAlloc::from(prov.0) {
                    if constant.ty().kind().is_raw_ptr() {
                        // If a static is a raw pointer, need to mark it as initialized.
                        self.push_target(MemoryInitOp::Set {
                            operand: Operand::Constant(constant.clone()),
                            value: true,
                            position: InsertPosition::Before,
                        });
                    }
                };
            }
        }
    }
    self.super_operand(operand, location);
}
```
It does consider the type of the static and only blesses raw pointers

Reply:
What do you mean by "blessing" ? Is it synonymous for "checking" ?

Reply:
By "blessing" I mean marking it as initialized in that case -- the opposite for checking. Sorry for the confusion!

Reply:
Let me add a check here.

Reply:
@remi-delmas-3000 I added a test to ensure that uninitialized checks correctly identifies the poison value in the static padding.